arcaflow_plugin_rtla/rtla_plugin.py

```diff
@@ -181,7 +213,67 @@ def run_timerlat(
                 total_usr_latency[label] = line_list[3]
 
         # Provide the rtla command formatted data as debug output
-        print(output)
+        print(timerlat_output)
+
+        # Example time series output from the tracer (truncated)
+        # <idle>-0      [009] d. 123.769498: #1002 context    irq timer_latency  458 ns
+        #  <...>-625890 [009] .. 123.769499: #1002 context thread timer_latency 666 ns
+        #  <...>-625890 [009] .. 123.769500: #1002 context user-ret timer_latency 499 ns
+        # <idle>-0      [002] d. 123.769528: #1003 context    irq timer_latency  587 ns
+        #  <...>-625883 [002] .. 123.769532: #1003 context thread timer_latency 712 ns
+        #  <...>-625883 [002] .. 123.769534: #1003 context user-ret timer_latency 462 ns
+
+        if params.enable_time_series and trace_file_exists:
+            timeseries_lines = iter(timeseries_output.splitlines())
+
+            # The calculation of the offset from uptime to current time certainly means
+            # that our time series is not 100% accurately aligned to the nanosecond, but
+            # the relative times will be accurate. We'll accept this as good enough.
+            uptime_offset = time.time() - time.monotonic()
+
+            if all(False for _ in timeseries_lines):
+                print(
+                    "No results reading tracer output; "
+                    "Skipping time series collection\n"
+                )
+
+            for line in timeseries_lines:
+                line_list = line.split()
+                # Because the tracer format is dependent on the underlying OS and cannot
+                # be controlled by the container, check the tracer output format and
+                # break gracefully if we don't recognize it
+                try:
+                    cpu = int(line_list[1][1:-1])
+                    uptimestamp = float(line_list[3][:-1])
+                    timestamp = str(
+                        datetime.fromtimestamp(uptimestamp + uptime_offset)
+                        .astimezone()
+                        .isoformat()
+                    )
```

Comment:
What do you think `astimezone()` is doing for you, here?  If you're trying to produce a timezone-aware `datetime` value, perhaps you should specify the `tz` parameter on the `fromtimestamp()` call?

Reply:
The advantage of `.astimezone()` is that it defaults to local time without having to supply a parameter, I suppose. Specifying an explicit zone in `fromtimestamp` would be a bit better documentation.

Reply:
Right but we have the whole provenience problem:  what timezone was the `uptimestamp` in?...what timezone are we trying to display it in?...how is using timezone-naive times (even briefly) going to relieve the confusion?  Wouldn't it be better to be explicit through the whole chain?